### PR TITLE
Ermahgerd Transform

### PR DESCRIPTION
**Pull request by stephenlb**
Transformation Option!

> return top 10 items sorted.

```javascript
eon.chart({
    channel   : 'ermahgerd',
    transform : function(m) {
        var columns = PUBNUB.map( m['top_items'], function( k, v ) {
            return [k, v.s+v.p+v.h];
        } );
        columns.sort(function( a, b ){ return a[1] < b[1] })
        columns.unshift(['x', new Date().getTime()]);
        return { columns: columns.slice(0,10) };
    }
});
```

### CODE REVIEW

File: pubnub-c3.js [hunk 1]
```diff
@@ -59,6 +57,8 @@ eon.c = {
       error = "PubNub not found. See http://www.pubnub.com/docs/javascript/javascript-sdk.html#_where_do_i_get_the_code";
     }
 
+    self.pubnub.options = options;
```
Comment (ianjennings) [inline pubnub-c3.js]: why?

File: pubnub-c3.js [hunk 2]
```diff
@@ -22,9 +22,7 @@ eon.c = {
         channel: channel,
         connect: connect,
         message: function(message, env, channel) {
-
-          eon.c.message(message, env, channel);
-
+          eon.c.message( pubnub.options.transform(message), env, channel );
```
Comment (ianjennings) [inline pubnub-c3.js]: why is this a property of ```pubnub``` and not just ```options```?

File: pubnub-c3.js (hunk 1 above)
Comment (stephenlb) [inline pubnub-c3.js]: to keep `transform()` method in scope.  May be a cleaner way

File: pubnub-c3.js (hunk 2 above)
Comment (stephenlb) [inline pubnub-c3.js]: The alternative is to modify the subscribe call and add in a new parameter.

```
eon.c.subscribe(self.pubnub, options.channel, options.connect, function(message, env, channel) {
```

**Pull request merged**